acceptance/openstack/identity/v3/identity.go

```diff
@@ -166,3 +182,71 @@ func DeleteDomain(t *testing.T, client *gophercloud.ServiceClient, domainID stri
 
 	t.Logf("Deleted domain: %s", domainID)
 }
+
+// UnassignRoleFromUserOnProject will revoke a role of a user on a project. A fatal error will
+// occur if the revoke was unsuccessful. This works best when used as a deferred function
+func UnassignRoleFromUserOnProject(t *testing.T, client *gophercloud.ServiceClient, role *roles.Role, user *users.User, project *projects.Project) {
+	t.Logf("Attempting to remove role %s from user %s on project %s", role.Name, user.Name, project.Name)
+
+	err := roles.UnassignFromUserOnProject(client, role.ID, user.ID, project.ID).ExtractErr()
+	if err != nil {
+		t.Fatalf("Unable to remove role")
+	}
+
+	t.Logf("Removed role %s from user %s on project %s", role.Name, user.Name, project.Name)
+}
+
+// FindRole finds all roles that the current authenticated client has access
+// to and returns the first one found. An error will be returned if the lookup
+// was unsuccessful.
+func FindRole(t *testing.T, client *gophercloud.ServiceClient) (*roles.Role, error) {
+	t.Log("Attempting to find a role")
+	var role *roles.Role
+
+	allPages, err := roles.List(client, nil).AllPages()
+	if err != nil {
+		return nil, err
+	}
+
+	allRoles, err := roles.ExtractRoles(allPages)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range allRoles {
+		role = &r
```

Comment:
Is allRoles stable, i.e. does the list contain the same ordered set of roles each time this gets called?